src/query/Index.cpp

```diff
@@ -91,14 +75,84 @@ std::vector<TriangleMatch> Index::getClosestTriangles(const mesh::Vertex &source
   }
 
   std::vector<TriangleMatch> matches;
-  _pimpl->indices.triangleRTree->query(bgi::nearest(sourceVertex, n),
+  _pimpl->indices.triangleRTree->query(bgi::nearest(sourceCoord, n),
                                        boost::make_function_output_iterator([&](impl::TriangleTraits::IndexType const &match) {
-                                         matches.emplace_back(bg::distance(sourceVertex, _mesh->triangles()[match.second]), match.second);
+                                         matches.emplace_back(bg::distance(sourceCoord, _mesh->triangles()[match.second]), match.second);
                                        }));
   std::sort(matches.begin(), matches.end());
   return matches;
 }
 
+VertexMatch Index::getClosestVertex(const mesh::Vertex &sourceVertex)
+{
+  PRECICE_TRACE();
+  auto matches = getClosestVertices(sourceVertex.getCoords(), 1);
+  // calling .back() on an empty vector is undefined.
+  if (not matches.empty()) {
+    return matches.back();
+  } else {
+    // return a default match with empty index
+    return VertexMatch(-1, std::numeric_limits<double>::max());
+  }
+}
+
+VertexMatch Index::getClosestVertex(const Eigen::VectorXd &sourceCoord)
```

Comment:
There is a more straight-forward implementation for this exact case, as the nearest neighbour is always the nearest projection for vertices.

Have a look at the old implementation:

https://github.com/precice/precice/blob/6fdeb97c20a92ecc47f55bdb00a8f95cc60b77af/src/mapping/NearestProjectionMapping.cpp#L141-L146